### PR TITLE
Pm 2022 07 lint rebase

### PR DESCRIPTION
**Pull request by phillimorland**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
     "eslint-plugin-jest": "26.5.3",
     "eslint-plugin-react": "7.30.1",
     "eslint-plugin-react-native": "4.0.0",
+    "eslint-plugin-react-native-a11y": "3.2.1",
```
Comment (goapunk) [inline package.json:78]: this officially only supports eslint 7 but we are using 8, therefore `make install` fails. I guess the two options we have are 
1. downgrade eslint
2.  force it to use eslint 8 and live with the risk
You can add this to package.json to make it work
```
 "overrides": {
    "eslint-plugin-react-native-a11y": {
      "eslint": "$eslint"
    }
  },
```

File: .eslintrc.json [hunk 2]
```diff
@@ -25,15 +25,15 @@
...
-    // General ESLINT rules
-    "indent": ["error", 2, { "SwitchCase": 1 }],
-    "linebreak-style": ["error", "unix"],
-    "quotes": ["error", "single"],
-    "semi": ["error", "always"],
-    "no-unused-vars": ["warn", { "varsIgnorePattern": "(React)" }],
-    "eol-last": ["error", "always"],
-    "no-multiple-empty-lines": ["error", { "max": 1 }],
-    "brace-style": ["error", "stroustrup", { "allowSingleLine": true }],
+      // General ESLINT rules
```
Comment (khamui) [inline .eslintrc.json:32]: somehow these rules are indented more than the following lines?

File: services/DateService.js [hunk 3]
```diff
@@ -11,18 +11,16 @@ const MONTHSTRINGS = {
...
 
 export const DateService = (dateInput, format = 'month d,y') => {
-  const d = new Date(dateInput);
+  const d = new Date(dateInput)
   if (format === 'month d,y') {
-    return `${MONTHSTRINGS[d.getMonth()]} ${d.getDate()}, ${d.getFullYear()}`;
+    return `${MONTHSTRINGS[d.getMonth()]} ${d.getDate()}, ${d.getFullYear()}`
+  } else if (format === 'month d, y, h:m') {
+    const hasMinutes = d.getMinutes() ? ':' + d.getMinutes() : ':00'
+    return `${MONTHSTRINGS[d.getMonth()]} ${d.getDate()}, ${d.getFullYear()}, ${d.getHours()}${hasMinutes}`
```
Comment (khamui) [inline services/DateService.js:22]: line too long?

**Review by khamui — APPROVED** [review]
really nice!! just two tiny comments

File: services/DateService.js (hunk 3 above)
Comment (phillimorland) [reply services/DateService.js:22]: yeah I didn't actually shorten all lines if it didn't seem necessary/worth it, when `make lint` there's still a bunch of warnings, do you think it's required here?

File: .eslintrc.json (hunk 2 above)
Comment (phillimorland) [reply .eslintrc.json]: fixed in later commit as rebasing issues

**Review by goapunk — APPROVED** [review]
really cool, thanks!

File: containers/Ideas/Phase.js [hunk 4]
```diff
@@ -27,25 +37,11 @@ export const Phase = props => {
...
-            <TextSourceSans>
-              It starts on {startDate}.
-            </TextSourceSans>
-          </View>
-        )}
+        <View>
+          <TextSourceSans>
+            {getInactivePhaseText()}
+          </TextSourceSans>
+        </View>
```
Comment (phillimorland) [inline containers/Ideas/Phase.js:44]: Nice solution! :)

Comment (Rineee) [reply containers/Ideas/Phase.js:44]: Yes, much better, thanks!

**Pull request merged**